elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/FutureImplementation.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+import com.yahoo.elide.datastores.aggregation.query.QueryResult;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.query.SQLQueryEngine;
+
+import java.util.concurrent.Future;
+
+public abstract class FutureImplementation implements Future<QueryResult> {
```

Comment:
You could make this an inner class of the SQLQueryEngine - not abstract - and it can just call the TransactionCancel member.